src/EditorFeatures/CSharpTest/CodeActions/InlineTemporary/InlineTemporaryTests.cs

```diff
@@ -4633,6 +4633,32 @@ void M()
     {
         Console.Write((Exception)null == new Exception());
     }
+}");
+        }
+
+        [WorkItem(30903, "https://github.com/dotnet/roslyn/issues/30903")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTemporary)]
+        public async Task InlineVariableContainsAliasOfValueTupleType()
+        {
+            await TestInRegularAndScriptAsync(
+@"using X = System.ValueTuple<int, int>;
+
+class C
+{
+    void M()
+    {
+        var [|x|] = (X)(0, 0);
+        var x2 = x;
+    }
+}",
+@"using X = System.ValueTuple<int, int>;
+
+class C
+{
+    void M()
+    {
+        var x2 = (System.ValueTuple<int, int>)(0, 0);
```

Comment:
i don't understand why this is the desired result.  why is it not:  `var x2 = (X)(0, 0);`?

Reply:
The cause of this is expander replaces `X` with `TupleTypeSyntax`, i.e. (int, int), instead of the expression in the using alias, i.e. `System.ValueTuple<int, int>`.

Although this seems to be a general issue when the expression in the using alias is different from whatever being simplified, this behavior is probably desired in all other cases. For example:
```cs
using Y = String;

void M()
{
    System.String y = "";  // provide simplification to `string y = ""`
    string z = "";              // provide simplification to `Y z = ""`
}
```

Reply:
@jcouv So if I understand this scenario correctly, we might need a symbol display option for ValueTuple that output in fully qualified name form, i.e. "System.ValueTuple<int, int>" instead of (int, int). Does this already exist?

Reply:
Looks like we have one, but it's internal. Is it accessible from IDE?
`     if (!format.CompilerInternalOptions.IncludesOption(SymbolDisplayCompilerInternalOptions.UseValueTuple))`

Reply:
Otherwise, you can use `INamedTypeSymbol.TupleUnderlyingType` to get the underlying type of the tuple (ie. `System.ValueTuple...`)

Reply:
> The cause of this is expander replaces X with TupleTypeSyntax, i.e. `(int, int)`, instead of the expression in the using alias, i.e. `System.ValueTuple<int, int>`.

That's appropriate.  First, tuple syntax is not ambiguous, so we can *should* be using tuple syntax *where legal*.  Second, we need tuple syntax so we can preserve names.  If we use `ValueTuple...` we will lose the names.  That's bad.  We should absolutely be using tuple syntax when it is legal to do so.

Reply:
> We should absolutely be using tuple syntax when it is legal to do so.

This is not true if the syntax we are expanding is an alias for a tuple type (i.e. this bug), since tuple type is not allowed in using alias.

Reply:
Inlining a local would cause the expression to be expanded and then later reduced. The cause of this is when a tuple type alias in the expression is being expanded, its tuple type form is used in stead of underlying type form. The fix takes advantage of the fact that tuple type form is not allowed in using alias.